exec/session.go

```diff
@@ -335,6 +338,21 @@ func (r *Result) Scanner() *sliceio.Scanner {
 	return sliceio.NewScanner(r, reader)
 }
 
+// Scope returns the merged metrics scope for the entire task graph represented
+// by the result r. Scope relies on the local values in the scopes of the task
+// graph, and thus are not precise.
+//
+// TODO(marius): flow and merge scopes along with data to provide precise
+// metrics.
+func (r *Result) Scope() *metrics.Scope {
+	r.initScope.Do(func() {
+		iterTasks(r.tasks, func(task *Task) {
```

Comment:
is merging incremental? This makes me think not, which is totally fine for now, but in the longer term you may want to merge all the scopes on a worker before shipping back to the master.

Reply:
it's not, but that would be possible. currently it's a lot simpler to do it this way, since we just have a scope associated with each task, that is updated on task completion... but (as i outlined in comments elsewhere), i think i'd eventually like this to follow the slice data flow instead, at which point incrementally would be "natural."